geoportal/c2cgeoportal_geoportal/scripts/c2cupgrade.py

```diff
@@ -276,14 +276,12 @@ def step0(self, step):
     def step1(self, step):
         check_call(["git", "reset", "--hard"])
         check_call(["git", "clean", "--force", "-d"])
```

Comment:
Is it possible to propose two choices here :  stash all (including untracked files) or cancel
Before running `git reset --hard` and `git clean --force -d`.

Reply:
We already have a message for that on step0, isn't clear enough?

Reply:
I see that there is also a status check at end of step0, can we move it at start of step1.

Reply:
I don't understand the goal of your purpose...

Reply:
To not execute that type of commands in user repo, and leave the user clear his working dir.

Reply:
> Here is the output of 'git status'. Please make sure to commit all your "
> "changes before going further. All uncommited changes will be lost.

Last time I've read this, I've stashed all my changes to tracked files, but I've lost all my untracked files.

Note that since I use git, I've never run `git clean -fd`. And when I upgrade some c2cgeoportal app, I manually add all modified files, because I sometime have to change things in const files. This is the developper job. Don't you do like this ? C2cgeoportal is only a framework, not the application itself, this is why we have a git repo in the customer app.

Reply:
Why not, I think that we shouldn't do that in this pull request, I purpose to create a Jira card for that...

Reply:
> This is the developper job.

The upgrade tool is made to be run by an integrator, not necessary a developer...

> C2cgeoportal is only a framework

Clearly not, it's an application builder... than more an application itself than a framework...

But it don't change the initial discussion, we need that to be able to do the final commit. if we don't do that (eventually manually) the changes (including new files) will be mixed up with the previous changes...

Reply:
I would prefer having them mixed than lost.